hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/scm/proxy/SCMSecurityProtocolFailoverProxyProvider.java

```diff
@@ -289,7 +289,7 @@ public synchronized void close() throws IOException {
     for (Map.Entry<String, ProxyInfo<SCMSecurityProtocolPB>> proxy :
         scmProxies.entrySet()) {
       if (proxy.getValue() != null) {
-        RPC.stopProxy(proxy.getValue());
+        RPC.stopProxy(proxy.getValue().proxy);
```

Comment:
why not just have an accessor for `proxy`

Reply:
I am not sure why this is needed

Reply:
Because otherwise, this code throws an error. The error is hidden Previously because `SCMSecurityProtocolFailoverProxyProvider` is never closed until the service closes.

Reply:
Let's file a jira and fix the behavior upstream.

Reply:
Ok, filed a jira upstream HDDS-8570.